Deprecate Python 2.7 and 3.5 support (#310)
twisted/treq: changelog.d/306.removal.rst

> Support for Python 3.5, which has reached end of support, is deprecated and will be dropped in the next treq release.

Reviewer: Maybe

```suggestion
Support for Python 3.5, which has reached end of support, is deprecated. This is the last release with support for Python 3.5.
```
